ako-gateway-api/k8s/gateway_controller.go

```diff
@@ -712,3 +728,26 @@ func validateAviConfigMap(obj interface{}) (*corev1.ConfigMap, bool) {
 	}
 	return nil, false
 }
+
+func validateReferreedHTTPRoute(key string, gateway *gatewayv1.Gateway, allowedRoutesAll bool) ([]*gatewayv1.HTTPRoute, error) {
+	namespace := gateway.Namespace
+	if allowedRoutesAll {
+		namespace = metav1.NamespaceAll
+	}
+	hrObj, err := akogatewayapilib.AKOControlConfig().GatewayApiInformers().HTTPRouteInformer.Lister().HTTPRoutes(namespace).List(labels.Set(nil).AsSelector())
+	httpRoutes := make([]*gatewayv1.HTTPRoute, 0)
+	if err != nil {
+		return nil, err
+	}
+	for _, httpRoute := range hrObj {
+		for _, pr := range httpRoute.Spec.ParentRefs {
+			if pr.Name == gatewayv1.ObjectName(gateway.Name) {
+				if IsHTTPRouteValid(key, httpRoute, gateway) {
+					httpRoutes = append(httpRoutes, httpRoute)
+				}
+				break
+			}
+		}
+	}
+	return httpRoutes, nil
```

Comment:
this list should be sorted before returning

Reply:
added sorting